### PR TITLE
Update packages for node security issues

### PR DESCRIPTION
**Pull request by carterbancroft**

### CODE REVIEW

**Comment by carterbancroft** [comment]
@shakefu done! I also added mongodb 4.0.x, but not 3.4 and 3.6. Let me know if you'd like me to add those as well.

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
   },
   "scripts": {
     "pretest": "./dbinit.sh ${AUTH_FLAG} init ${MONGODB_VERSION}",
-    "test": "coffeelint index.coffee test/ && mocha --require coffee-script/register --compilers coffee:coffee-script/register",
+    "test": "coffeelint index.coffee test/ && mocha --require coffee-script/register 'test/**/*.coffee'",
```
Comment (carterbancroft) [inline package.json]: Changed this after updating mocha based on the deprecation of --compilers (see: https://github.com/mochajs/mocha/wiki/compilers-deprecation)

File: test/index.coffee [hunk 2]
```diff
@@ -169,13 +169,13 @@ describe 'Document', ->
     cursor = MyDoc.collection.find {}
     cursor.limit 1, (err, docs) ->
       throw err if err
-      docs.should.have.length.of 1
+      docs.should.have.lengthOf 1
```
Comment (carterbancroft) [inline test/index.coffee]: It seems like Chai syntax has changed in later versions. Updated to fix tests after upgrading chai.

File: .travis.yml [hunk 3]
```diff
@@ -3,10 +3,14 @@ node_js:
...
+  - "13"
 env:
   - MONGODB_VERSION=2.6.x AUTH_FLAG='-a'
   - MONGODB_VERSION=2.6.x
   - MONGODB_VERSION=3.0.x AUTH_FLAG='-a'
   - MONGODB_VERSION=3.0.x
   - MONGODB_VERSION=3.2.x AUTH_FLAG='-a'
   - MONGODB_VERSION=3.2.x
+  - MONGODB_VERSION=4.0.x AUTH_FLAG='-a'
+  - MONGODB_VERSION=4.0.x
```
Comment (carterbancroft) [inline .travis.yml]: Only added v4 here, but lemme know if you want 3.4 and 3.6 as well.

File: .travis.yml [hunk 4]
```diff
@@ -1,12 +1,15 @@
 language: node_js
 node_js:
-  - "6"
```
Comment (carterbancroft) [inline .travis.yml]: Tests fail in node 6, I think it's because of updating mocha but maybe we can just drop node 6 support here?

Comment (NigelKibodeaux) [reply .travis.yml]: i don't object

**Pull request merged**